### PR TITLE
Use the AEM Testing Clients for integration tests

### PR DESCRIPTION
**Pull request by francescomari**
This PR replaces the content of the it.tests module with some sample integration tests based on the [AEM Testing Clients](https://github.com/adobe/aem-testing-clients).

## Description

The changes performed are the following:
* Delete the old content of `it.tests`.
* Add sample integration tests in `it.tests` based on the AEM Testing Client. The tests assert that it is possible to create as well as read pages.
* Remove the `it.launcher` module, that is not needed for the new `it.tests`.
* Select the correct version of the AEM Testing Clients depending on the `aemVersion` used to generate the project.

## Motivation and Context

The server-side tests implemented by the current version of it.tests are not supported for cloud deployments of AEM. The correct way forward for cloud AEM deployments is to write integration tests based on the AEM Testing Clients. Moreover, the AEM Testing Clients are released depending on the AEM version they target, making them a perfect fit for this archetype.

## How Has This Been Tested?

I've run the tests for the archetype, and I manually tested the generation of an archetype for AEM 6.3 and 6.4, which are not currently covered by the build.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Comment by francescomari** [comment]
@bpauli I think the PR is in a reviewable state. Please let me know if I missed something!

**Comment by kwin** [comment]
Just for the record. This would fix #354

**Comment by msagolj** [comment]
@francescomari can you also please update the README of the generated archetype and provide an example on how to execute the tests.

**Comment by francescomari** [comment]
@msagolj, I added documentation about the integration tests at d5eddcf.

File: src/main/archetype/it.tests/pom.xml [hunk 1]
```diff
@@ -17,138 +17,194 @@
...
+            <artifactId>junit</artifactId>
+            <version>4.13</version>
         </dependency>
+#if ($aemVersion == "cloud")
         <dependency>
-            <groupId>org.osgi</groupId>
-            <artifactId>org.osgi.annotation.bundle</artifactId>
+            <groupId>com.adobe.cq</groupId>
+            <artifactId>aem-cloud-testing-clients</artifactId>
+            <version>0.2.3</version>
```
Comment (volteanu) [inline src/main/archetype/it.tests/pom.xml:47]: we already have `0.2.4` 😄

File: src/main/archetype/it.tests/pom.xml [hunk 2]
```diff
@@ -17,138 +17,194 @@
...
+            <properties>
+                <!-- Cloud author service (LB) -->
+                <cloud.author.url>http://localhost:4502</cloud.author.url>
+                <cloud.author.user>admin</cloud.author.user>
+                <cloud.author.password>admin</cloud.author.password>
+
+                <!-- cloud publish service (LB) -->
+                <cloud.publish.url>http://localhost:4503</cloud.publish.url>
+                <cloud.publish.user>admin</cloud.publish.user>
+                <cloud.publish.password>admin</cloud.publish.password>
```
Comment (volteanu) [inline src/main/archetype/it.tests/pom.xml:87]: wondering if the names of these properties aren't too confusing (local profile - cloud properties?!). What about we go with `it.author.url` and so forth (`s/cloud/it`)?

File: src/main/archetype/it.tests/pom.xml (hunk 1 above)
Comment (francescomari) [reply src/main/archetype/it.tests/pom.xml]: Upgraded at bc4acf8.

File: src/main/archetype/it.tests/pom.xml (hunk 2 above)
Comment (francescomari) [reply src/main/archetype/it.tests/pom.xml]: Fixed at 84bdef7.

File: src/main/archetype/it.tests/pom.xml [hunk 3]
```diff
@@ -17,138 +17,194 @@
...
+                        </configuration>
+                        <executions>
+                            <execution>
+                                <id>default-integration-test</id>
+                                <goals>
+                                    <goal>integration-test</goal>
+                                </goals>
+                                <configuration>
+                                    <includes>
+                                        <include>**/*IT.java</include>
```
Comment (volteanu) [inline src/main/archetype/it.tests/pom.xml:138]: I think this is an important convention to be documented.

File: src/main/archetype/it.tests/pom.xml [hunk 4]
```diff
@@ -17,138 +17,194 @@
...
+        </pluginManagement>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-assembly-plugin</artifactId>
+                <version>3.1.0</version>
+                <configuration>
+                    <descriptorRefs>
+                        <descriptorRef>jar-with-dependencies</descriptorRef>
+                    </descriptorRefs>
```
Comment (volteanu) [inline src/main/archetype/it.tests/pom.xml:196]: according to [Cloud Manager docs](https://docs.adobe.com/content/help/en/experience-manager-cloud-service/implementing/using-cloud-manager/understand-test-results.html)  the JAR must have the Cloud-Manager-TestType manifest header set to integration-test.

File: src/main/archetype/it.tests/src/main/java/it/tests/CreatePageIT.java [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+import com.adobe.cq.testing.junit.rules.CQRule;
+import com.adobe.cq.testing.junit.rules.Page;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+
+public class CreatePageIT {
```
Comment (volteanu) [inline src/main/archetype/it.tests/src/main/java/it/tests/CreatePageIT.java:30]: I would add exhaustive javadocs for this class to help the reader understand every line and it's purpose.

File: src/main/archetype/it.tests/pom.xml (hunk 4 above)
Comment (francescomari) [reply src/main/archetype/it.tests/pom.xml:196]: Fixed at 46a268c.

File: src/main/archetype/it.tests/pom.xml (hunk 3 above)
Comment (francescomari) [reply src/main/archetype/it.tests/pom.xml:138]: Fixed at cb8c46d.

File: src/main/archetype/it.tests/src/main/java/it/tests/CreatePageIT.java (hunk 5 above)
Comment (francescomari) [reply src/main/archetype/it.tests/src/main/java/it/tests/CreatePageIT.java:37]: Fixed at 49526ac.

**Comment by francescomari** [comment]
@volteanu, I documented the tests at 49526ac to the best of my knowledge of the AEM Testing Clients. Please have a look for potential inaccuracies.

**Comment by volteanu** [comment]
Thanks @francescomari for tackling the docs, looks super good 💯.

BTW, I've tried the archetype and the it module looks good to me, was able to build and run the tests!

File: src/main/archetype/it.tests/src/main/java/it/tests/GetPageIT.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    public CQRule cqBaseRule = new CQRule(cqBaseClassRule.authorRule, cqBaseClassRule.publishRule);
+
+    // Page will create a test page with a random name and it will make sure
+    // that the page is removed at the end of every test execution. By using a
+    // random name, your test will not conflict with any other test running on
+    // the same instance. By removing the page at the end of the test execution,
+    // you are not going to leave any clutter on the instance under test.
+
+    @Rule
+    public Page root = new Page(cqBaseClassRule.authorRule);
```
Comment (dulvac) [inline src/main/archetype/it.tests/src/main/java/it/tests/GetPageIT.java:57]: This one is not needed for this test class and would create and delete a page for every test that just does a GET

Comment (francescomari) [reply src/main/archetype/it.tests/src/main/java/it/tests/GetPageIT.java]: Fixed at 202fcd8.

**Pull request merged**